src/control/build/variables.go

```diff
@@ -27,3 +33,26 @@ var (
 	// DefaultSystemName defines the default DAOS system name.
 	DefaultSystemName = "daos_server"
 )
+
+func revString() string {
+	var revString string
+	if info, ok := debug.ReadBuildInfo(); ok {
+		for _, setting := range info.Settings {
```

Comment:
@mjmac Any reason why I get this error when trying to build locally?
```
# github.com/daos-stack/daos/src/control/build
build/variables.go:40:31: info.Settings undefined (type *debug.BuildInfo has no field or method Settings)
```
Using:
`scons-3 --build-deps=yes  --config=force TARGET_TYPE=dev BUILD_TYPE=dev install -j68`

Reply:
```
[dbohning@boro-27 ~]$ go version
go version go1.17.7 linux/amd64
```

Reply:
Thanks. I'll try upgrading

Reply:
It seems go 1.17 is the latest version available on my node. Maybe because it's EL 8.5? I'll try an 8.6 node, but I do wonder if others will have similar problems

Reply:
Even my 8.6 node (for example, boro-24) only has go 1.17 available. Are we building/using a special go RPM in CI?

Reply:
1.19.4 is available for appstream-8-stream, but not 8.6/8.7
And it doesn't seem the daos deps repo has it either:
```
[dbohning@boro-24 ~]$ sudo dnf install go
...
go    x86_64    1.17.7-1.daos.el8    daos-stack-deps-el-8-x86_64-stable-local-artifactory
```

Reply:
Ah, sorry. It's "golang". That old 1.17.7 RPM is cruft from a long time ago.
```
[macdonal@wolf-61 daos]$ sudo dnf info golang
Last metadata expiration check: 1:11:15 ago on Thu 27 Apr 2023 01:47:52 PM UTC.
Installed Packages
Name         : golang
Version      : 1.20.3
Release      : 1.daos.el8
Architecture : x86_64
Size         : 218 M
Source       : golang-1.20.3-1.daos.el8.src.rpm
Repository   : @System
From repo    : daos-stack-deps-el-8-x86_64-stable-local-artifactory
Summary      : The Go Programming Language
URL          : http://golang.org/
License      : BSD and Public Domain
Description  : Installs the precompiled Go toolchain provided at https://go.dev/dl/, replacing
             : any older distro-provided versions.
```

Reply:
That's not available either
```
[dbohning@boro-24 ~]$ sudo dnf list golang
Last metadata expiration check: 3:27:27 ago on Thu 27 Apr 2023 11:43:16 AM UTC.
Available Packages
golang.x86_64    1.17.12-1.module+el8.6.0+993+4fde2340                                                                            daos_ci-rocky8-appstream-artifactory

[dbohning@boro-24 ~]$ sudo dnf list go
Last metadata expiration check: 3:27:34 ago on Thu 27 Apr 2023 11:43:16 AM UTC.
Available Packages
go.src       1.17.7-1.daos.el8    daos-stack-deps-el-8-x86_64-stable-local-artifactory
go.x86_64    1.17.7-1.daos.el8    daos-stack-deps-el-8-x86_64-stable-local-artifactory
```

Reply:
I'm not sure what to say... Seems like something wonky with your node's yum/dnf config. We use the go 1.20.3 RPM in CI all the time.

```
[macdonal@wolf-61 daos]$ sudo yum repolist -v
Loaded plugins: builddep, changelog, config-manager, copr, debug, debuginfo-install, download, generate_completion_cache, groups-manager, needs-restarting, playground, repoclosure, repodiff, repograph, repomanage, reposync
YUM version: 4.7.0
cachedir: /var/tmp/dnf-macdonal-lzn4hu1g
Repo-id            : appstream
Repo-name          : Rocky Linux 8.7 - AppStream
Repo-revision      : 8.7
Repo-distro-tags      : [cpe:/o:rocky:rocky:8]:  ,  , 8, L, R, c, i, k, n, o, u, x, y
Repo-updated       : Wed 26 Apr 2023 03:44:31 PM UTC
Repo-pkgs          : 7,828
Repo-available-pkgs: 6,299
Repo-size          : 17 G
Repo-mirrors       : https://mirrors.rockylinux.org/mirrorlist?arch=x86_64&repo=AppStream-8.7
Repo-baseurl       : http://iad.mirror.rackspace.com/rocky/8.7/AppStream/x86_64/os/ (47 more)
Repo-expire        : 172,800 second(s) (last: Thu 27 Apr 2023 03:26:37 PM UTC)
Repo-filename      : /etc/yum.repos.d/Rocky-AppStream.repo

Repo-id            : baseos
Repo-name          : Rocky Linux 8.7 - BaseOS
Repo-revision      : 8.7
Repo-distro-tags      : [cpe:/o:rocky:rocky:8]:  ,  , 8, L, R, c, i, k, n, o, u, x, y
Repo-updated       : Wed 26 Apr 2023 03:44:36 PM UTC
Repo-pkgs          : 2,195
Repo-available-pkgs: 2,193
Repo-size          : 3.0 G
Repo-mirrors       : https://mirrors.rockylinux.org/mirrorlist?arch=x86_64&repo=BaseOS-8.7
Repo-baseurl       : http://iad.mirror.rackspace.com/rocky/8.7/BaseOS/x86_64/os/ (47 more)
Repo-expire        : 172,800 second(s) (last: Thu 27 Apr 2023 03:26:41 PM UTC)
Repo-filename      : /etc/yum.repos.d/Rocky-BaseOS.repo

Repo-id            : daos-stack-deps-el-8-x86_64-stable-local-artifactory
Repo-name          : daos-stack-deps-el-8-x86_64-stable-local-artifactory
Repo-updated       : Wed 26 Apr 2023 10:10:14 PM UTC
Repo-pkgs          : 1,356
Repo-available-pkgs: 1,356
Repo-size          : 4.5 G
Repo-baseurl       : https://artifactory.dc.hpdd.intel.com/artifactory/daos-stack-deps-el-8-x86_64-stable-local/
Repo-expire        : 172,800 second(s) (last: Thu 27 Apr 2023 03:26:47 PM UTC)
Repo-filename      : /etc/yum.repos.d/daos_ci-rocky8-artifactory.repo

Repo-id            : extras
Repo-name          : Rocky Linux 8.7 - Extras
Repo-revision      : 1682440000
Repo-distro-tags      : [cpe:/o:rocky:rocky:8]:  ,  , 8, L, R, c, i, k, n, o, u, x, y
Repo-updated       : Tue 25 Apr 2023 04:26:40 PM UTC
Repo-pkgs          : 57
Repo-available-pkgs: 57
Repo-size          : 5.5 M
Repo-mirrors       : https://mirrors.rockylinux.org/mirrorlist?arch=x86_64&repo=extras-8.7
Repo-baseurl       : http://iad.mirror.rackspace.com/rocky/8.7/extras/x86_64/os/ (47 more)
Repo-expire        : 172,800 second(s) (last: Thu 27 Apr 2023 03:26:43 PM UTC)
Repo-filename      : /etc/yum.repos.d/Rocky-Extras.repo

Repo-id            : gh-cli
Repo-name          : packages for the GitHub CLI
Repo-revision      : 1682430747
Repo-updated       : Tue 25 Apr 2023 01:52:27 PM UTC
Repo-pkgs          : 4
Repo-available-pkgs: 4
Repo-size          : 38 M
Repo-baseurl       : https://cli.github.com/packages/rpm
Repo-expire        : 172,800 second(s) (last: Thu 27 Apr 2023 03:26:48 PM UTC)
Repo-filename      : /etc/yum.repos.d/gh-cli.repo
Total packages: 11,440
```

Reply:
This is probably another discrepancy between CI images and the images available with nodemgr

Reply:
@brianjmurrell @JohnMalmberg Could you please weigh in on why our 8.6 images (specifically my boro-24 and boro-25 nodes) don't have this go package available?

Reply:
@daltonbohning  Not really.  There are too many variables.  Our build is done in Docker using `utils/docker/Dockerfile.el.8`.  But due to Docker's caching I suspect we cannot see what it did to get _go_ installed in any of the runs of this PR.

My suggestion would be to try to do your build with our Dockerfile, if for no other reason than to see the detail of the commands that it's executing (i.e. where from/how it installs _go_) that we cannot see in any runs of this PR due to the caching in Docker.  Feel free to paste your `docker run` output here if you are having trouble figuring out what it did.

It might also be that we have just come across our first need/reason to make 8.7 the minimum EL supported version.

If that is the case (and my research is indicating the same), @Michael-Hennecke will need to be brought into this discussion however as he is the gatekeeper of what versions of distros we support for each release.

We need to consider that if 8.7 is our new minimum release due to these changes, that would have become more clear sooner if we were not building with an out-of-distro _go_ packaging (which I realize we need to do for SDL).  Indeed, if you had not discovered it through this experimentation, I am not sure what would have helped us realize this new minimum.

@mjmac Any ideas on how our SDL requirement for using the latest _go_ doesn't paper over distribution minimum requirements?  Maybe another build stage that does not use our in-house packaging of the most recent _go_ that we ultimately discard at the end of the build perhaps?

Speaking of new stages, I feel like I'm seeing new stages being added to `Jenkinsfile` left-right-and-centre in various PRs these days.  I'm getting a bad feeling that we are going to blow up the `Jenkinsfile` limit pretty soon with a few landings.  Maybe this new stage is a good candidate for a GitHub Actions stage.

Reply:
@brianjmurrell I'm not familiar with docker, but I managed to run
```
sudo docker build . --file utils/docker/Dockerfile.el.8 --build-arg DAOS_BUILD=no --build-arg DEPS_JOBS --build-arg BASE_DISTRO --build-arg DAOS_KEEP_SRC=yes --tag build-image
```
It installed 1.18:
`golang    x86_64  1.18.10-1.module+el8.7.0+1176+40c90211 appstream`
Though I'm not sure where that came from or how to check. Learning docker to check this seems way overkill considering my lack of knowledge.

But even if CI/docker does **somehow** install v1.18 from our internal repos, we'd have to make sure those versions are available to users.

Reply:
> @brianjmurrell I'm not familiar with docker, but I managed to run
> 
> It installed 1.18: `golang x86_64 1.18.10-1.module+el8.7.0+1176+40c90211 appstream`

---------------------------------------------------------------------------------^^^
So that looks to me like it was because your Docker container used EL 8.7 which is likely what CI is doing at this point.

You could add a `cat /etc/os-release` at the end of that Dockerfile.el.8 to be sure.

You could then replace 
https://github.com/daos-stack/daos/blob/e64988e9e61a47ebe355d6c34af2c30e7875f178/utils/docker/Dockerfile.el.8#L9

with

```
ARG BASE_DISTRO=rockylinux/rockylinux:8.6
```

or maybe you'd need

```
ARG BASE_DISTRO=rockylinux/rockylinux:8.6.20227707
```

and try your build again and see what version of _go_ you get.

> But even if CI/docker does **somehow** install v1.18 from our internal repos, we'd have to make sure those versions are available to users.

Well that is just the problem.  We don't want to be getting into having to distribute toolchains just to be able to build DAOS.  That just ever increases our SDL footprint.  Per my last message, we need to specify the versions of the distro that build DAOS with their native toolchains.  That *seems* at this point, with this PR, to mean that we are bumping the minimum EL8 version to 8.7 due to the need for _go_ 1.18 now.